Mathlib/Analysis/Convex/Basic.lean

```diff
@@ -605,7 +605,8 @@ theorem convex_stdSimplex : Convex 𝕜 (stdSimplex 𝕜 ι) := by
 lemma stdSimplex_of_isEmpty_index [IsEmpty ι] [Nontrivial 𝕜] : stdSimplex 𝕜 ι = ∅ :=
   eq_empty_of_forall_not_mem <| by rintro f ⟨-, hf⟩; simp at hf
 
-lemma stdSimplex_unique [Unique ι] : stdSimplex 𝕜 ι = {fun _ ↦ 1} := by
+lemma stdSimplex_unique [Nonempty ι] [Subsingleton ι] : stdSimplex 𝕜 ι = {fun _ ↦ 1} := by
+  cases (unique_iff_subsingleton_and_nonempty ι).mpr ⟨‹_›, ‹_›⟩
```

Comment:
Can't you more directly use `nonempty_unique` here?

Reply:
The old proof goes via [Unique.fintype](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Basic.html#Unique.fintype) (which shouldn't be an instance, because it's not clear whether it creates a diamond with instances for `Prod` and pi types).

Reply:
Fixed. I added `nonempty_unique` after I already wrote most of the PR. Then I migrated most changes to use it, but forgot this one.